packages/cozy-client/src/models/file.js

```diff
@@ -291,3 +292,268 @@ export const isReferencedByAlbum = file => {
+
+/**
+ * async getFullpath - Gets a file's path
+ *
+ * @param {CozyClient} client - The CozyClient instance
+ * @param {string} dirId  - The id of the parent directory
+ * @param {string} name  - The file's name
+ * @returns {Promise<string>}       The full path of the file in the cozy
+ **/
+export const getFullpath = async (client, dirId, name) => {
+  if (!dirId) {
+    throw new Error('You must provide a dirId')
+  }
+
+  const { data: parentDir } = await client.query(
+    Q(DOCTYPE_FILES).getById(dirId)
+  )
+  const parentDirectoryPath = trimEnd(parentDir.path, '/')
+  return `${parentDirectoryPath}/${name}`
+}
+
+/**
+ * Move file to destination.
+ *
+ * @param {CozyClient} client - The CozyClient instance
+ * @param   {string} fileId               - The file's id (required)
+ * @param   {object} destination
+ * @param   {string} destination.folderId - The destination folder's id (required)
+ * @param   {string} destination.path     - The file's path after the move (optional, used to optimize performance in case of conflict)
+ * @param   {boolean} force                - Whether we should overwrite the destination in case of conflict (defaults to false)
+ * @returns {Promise}                     - A promise that returns the move action response and the deleted file id (if any) if resolved or an Error if rejected
+ *
+ */
+export const move = async (client, fileId, destination, force = false) => {
+  const { folderId, path } = destination
+  try {
+    const resp = await client
+      .collection(DOCTYPE_FILES)
+      .updateFileMetadata(fileId, {
+        dir_id: folderId
+      })
+
+    return {
+      moved: resp.data,
+      deleted: null
+    }
+  } catch (e) {
+    if (e.status === 409 && force) {
+      let destinationPath
+      if (path) {
+        destinationPath = path
+      } else {
+        const { data: movedFile } = await client.query(
+          Q(DOCTYPE_FILES).getById(fileId)
+        )
+        const filename = movedFile.name
+        destinationPath = await getFullpath(client, folderId, filename)
+      }
+      const conflictResp = await client
+        .collection(DOCTYPE_FILES)
+        .statByPath(destinationPath)
+      await client.collection(DOCTYPE_FILES).destroy(conflictResp.data)
+      const resp = await client
+        .collection(DOCTYPE_FILES)
+        .updateFileMetadata(fileId, {
+          dir_id: folderId
+        })
+
+      return {
+        moved: resp.data,
+        deleted: conflictResp.data.id
+      }
+    } else {
+      throw e
+    }
+  }
+}
+
+/**
+ *
+ * Method to upload a file even if a file with the same name already exists.
+ *
+ * @param {CozyClient} client - The CozyClient instance
+ * @param {string} pathArg Fullpath for the file ex: path/to/
+ * @param {object} file HTML Object file
+ * @param {Object} metadata An object containing the wanted metadata to attach
+ */
+export const overrideFileForPath = async (client, pathArg, file, metadata) => {
```

Comment:
May I suggest `overwrite` instead of `override`?

Reply:
As this method already exist in https://github.com/cozy/cozy-libs/blob/master/packages/cozy-doctypes/src/File.js#L101 , I feel like this will easier to stick with the same name for future migrations?